tests/functional/workloads/cnv/test_cnv_node_replacement.py

```diff
@@ -0,0 +1,140 @@
+import logging
+import pytest
+import random
+
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import E2ETest
+from ocs_ci.framework.pytest_customization.marks import workloads, magenta_squad
+from ocs_ci.helpers.cnv_helpers import cal_md5sum_vm, run_dd_io
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.cluster import CephCluster
+from ocs_ci.ocs.resources.storage_cluster import osd_encryption_verification
+from tests.functional.z_cluster.nodes.test_node_replacement_proactive import (
+    delete_and_create_osd_node,
+    select_osd_node_name,
+)
+from ocs_ci.helpers.helpers import (
+    verify_storagecluster_nodetopology,
+)
+from ocs_ci.helpers.sanity_helpers import Sanity
+
+logger = logging.getLogger(__name__)
+
+
+@magenta_squad
+@workloads
+class TestCnvNodeReplace(E2ETest):
+    """
+    Node replacement proactive
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, project_factory, multi_cnv_workload):
+        """
+        Setting up VMs for tests
+
+        """
+
+        # Create a project
+        proj_obj = project_factory()
+        (
+            self.vm_objs_def,
+            self.vm_objs_aggr,
+            self.sc_obj_def_compr,
+            self.sc_obj_aggressive,
+        ) = multi_cnv_workload(namespace=proj_obj.namespace)
+
+        logger.info("All vms created successfully")
+
+    def test_vms_with_node_replacement(
+        self,
+        setup_cnv,
+        setup,
+    ):
+        """
+        Node Replacement proactive
+        """
+        all_vms = self.vm_objs_def + self.vm_objs_aggr
+
+        file_paths = ["/source_file.txt", "/new_file.txt"]
+        source_csums = {}
+        for vm_obj in all_vms:
+            source_csum = run_dd_io(vm_obj=vm_obj, file_path=file_paths[0], verify=True)
+            source_csums[vm_obj.name] = source_csum
+
+        # Choose VMs randomaly
+        vm_for_clone, vm_for_stop, vm_for_snap = random.sample(all_vms, 3)
+
+        # Uncomment code ones 11199 merged.
+        """
+        # Create Clone of VM
+        cloned_vm = clone_or_snapshot_vm(
+            "clone",
+            vm_for_clone,
+            admin_client=admin_client,
+            all_vms=all_vms,
+            file_path=file_paths[0],
+        )
+        csum = cal_md5sum_vm(vm_obj=cloned_vm, file_path=file_paths[0])
+        source_csums[cloned_vm.name] = csum
+        # Create a snapshot
+        vm_for_snap = clone_or_snapshot_vm(
+            "snapshot", vm_for_snap, admin_client=admin_client, file_path=file_paths[0]
+        )
+        csum = cal_md5sum_vm(vm_obj=vm_for_snap, file_path=file_paths[0])
+        source_csums[vm_for_snap.name] = csum
+        """
+
+        # Keep vms in different states (power on, paused, stoped)
+        vm_for_stop.stop()
+        vm_for_snap.pause()
+
+        # Replace Node
+        osd_node_name = select_osd_node_name()
+        delete_and_create_osd_node(osd_node_name)
+
```

Comment:
post node replacement, you need to also make sure that the VM successfully migrates to another healthy node and also need to check the VM data integrity

Reply:
Done

Reply:
https://github.com/red-hat-storage/ocs-ci/pull/11376/files#diff-2d860c7ec34e7d3bc3c5b860cd619fe306a24294090ae17030a4f0039282586bR148